File: backend/app/schemas/analytics.py
from pydantic import BaseModel


class PnlTimeSeriesPoint(BaseModel):
    date: str
    pnl: float = 0.0
    cumulative_pnl: float = 0.0
    bets_count: int = 0
    wins: int = 0


class PnlByLeague(BaseModel):
    league: str
    total_pnl: float = 0.0
    bets_count: int = 0
    wins: int = 0
    win_rate: float = 0.0


class PnlByModel(BaseModel):
    model_type: str
    total_pnl: float = 0.0
    bets_count: int = 0
    wins: int = 0
    win_rate: float = 0.0


class EquityCurvePoint(BaseModel):
    date: str
    balance: float


class ClvLegResult(BaseModel):
    ticket_id: int
    ticket_leg_id: int
    reference_stage: str | None = None
    same_book_clv_pct: float | None = None
    market_best_clv_pct: float | None = None
    consensus_clv_pp: float | None = None
    coverage: dict[str, bool]
    unavailable_reasons: dict[str, tuple[str, ...]]


class ClvSummary(BaseModel):
    leg_count: int = 0
    same_book_coverage_pct: float = 0.0
    market_best_coverage_pct: float = 0.0
    consensus_coverage_pct: float = 0.0
    average_same_book_clv_pct: float | None = None
    average_market_best_clv_pct: float | None = None
    average_consensus_clv_pp: float | None = None
    positive_same_book_pct: float | None = None
    positive_market_best_pct: float | None = None
    positive_consensus_pct: float | None = None


class ClvReport(BaseModel):
    summary: ClvSummary
    items: list[ClvLegResult]

File: frontend/src/lib/api/analytics.ts
import { ApiClient } from './client';
import type { PnlPoint } from '$lib/types';

export type ClvReport = {
	summary: {
		leg_count: number;
		same_book_coverage_pct: number;
		market_best_coverage_pct: number;
		consensus_coverage_pct: number;
		average_same_book_clv_pct: number | null;
		average_market_best_clv_pct: number | null;
		average_consensus_clv_pp: number | null;
		positive_same_book_pct: number | null;
		positive_market_best_pct: number | null;
		positive_consensus_pct: number | null;
	};
	items: Array<{
		ticket_id: number;
		ticket_leg_id: number;
		reference_stage: 'generation' | 'activation' | null;
		same_book_clv_pct: number | null;
		market_best_clv_pct: number | null;
		consensus_clv_pp: number | null;
		coverage: Record<string, boolean>;
		unavailable_reasons: Record<string, string[]>;
	}>;
};

class AnalyticsApi extends ApiClient {
	async getPnl(period?: string, group_by?: string): Promise<PnlPoint[]> {
		const p = period || '30d';
		const g = group_by || 'day';
		return this.get<PnlPoint[]>(`/api/v1/analytics/pnl?period=${p}&group_by=${g}`);
	}

	async getPnlByLeague(): Promise<
		{ league: string; pnl: number; bets: number; win_rate: number }[]
	> {
		return this.get('/api/v1/analytics/pnl/by-league');
	}

	async getPnlByModel(): Promise<
		{ model: string; pnl: number; bets: number; win_rate: number }[]
	> {
		return this.get('/api/v1/analytics/pnl/by-model');
	}

	async getEquityCurve(period?: string): Promise<
		{ date: string; balance: number }[]
	> {
		const p = period || '30d';
		return this.get(`/api/v1/analytics/equity-curve?period=${p}`);
	}

	async getClv(): Promise<ClvReport> {
		return this.get('/api/v1/analytics/clv');
	}
}

export const analyticsApi = new AnalyticsApi();
